src/roles/transporter.ts

```diff
@@ -0,0 +1,76 @@
+import { RoleName } from "enums/RoleName";
+import { Tasks } from "../creep-tasks/Tasks";
+import { Role } from "./Role";
+
+export class RoleHarvester extends Role {
+    public static roleName: string = RoleName.Harvester;
+
+    public newTask(creep: Creep): void {
+        if (creep.carry.energy < creep.carryCapacity) {
+            // Find Structures for getting energy
+            const structures = creep.room.find(FIND_STRUCTURES);
+
+            // Withdraw from containers
+            const containersWithEnergy = structures.filter(s => s.structureType === STRUCTURE_CONTAINER && s.store[RESOURCE_ENERGY] > 0) as StructureContainer[];
+            if (containersWithEnergy.length > 0) {
+                creep.task = Tasks.withdraw(containersWithEnergy[0], RESOURCE_ENERGY);
+                return;
+            }
+
+            // Pickup dropped energy
+            const droppedEnergy = creep.pos.findClosestByPath(FIND_DROPPED_RESOURCES, { filter: (d) => d.resourceType === RESOURCE_ENERGY });
+            if (droppedEnergy != null) {
+                creep.task = Tasks.pickup(droppedEnergy);
+                return;
+            }
+
+            // Pull energy from storage
+            const storageWithEnergy = structures.filter(s => s.structureType === STRUCTURE_STORAGE && s.store[RESOURCE_ENERGY] > 0) as StructureStorage[];
+            if (storageWithEnergy.length > 0) {
+                creep.task = Tasks.withdraw(storageWithEnergy[0], RESOURCE_ENERGY);
+                return;
+            }
+
+            // If energy is not available at the time continue to transfer tasks (minimize idle time)
+            if (creep.carry.energy < creep.carryCapacity / 2) {
+                return;
```

Comment:
If you want it to continue, remove this return

Reply:
I don't want it to continue. That's why it's returning.